scripts/engineering/derive_phase44d_carry_aware_family_constant_surface.py

```
@@ -0,0 +1,466 @@
+    except ValueError as exc:
+        raise SystemExit(f"{label} must be numeric in {path}; got {raw!r}") from exc
+    if value < 0:
+        raise SystemExit(f"{label} must be non-negative in {path}; got {raw!r}")
+    return value
+
+
+def parse_int(raw: str, *, label: str, path: Path) -> int:
+    try:
+        return int(raw)
+    except ValueError as exc:
+        raise SystemExit(f"{label} must be an integer in {path}; got {raw!r}") from exc
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)